README.md

```diff
@@ -2,69 +2,82 @@
+  many sequences that might be useful test cases, generalized in
+  [Graft](https://github.com/tweag/graft)
+- deploy automated attacks over existing sequences of transactions,
+  such as datum hijacking or double satisfaction attacks, in an attempt
+  to uncover vulnerabilities
 
-The library is geared specifically towards testing and auditing on-chain code.
-
-You are free to copy, modify, and distribute `cooked-validators` under the terms
-of the MIT license. We provide `cooked-validators` as a research prototype under
-active development, and it comes _as is_ with no guarantees whatsoever. Check
-the [license](LICENSE) for details.
+You are free to copy, modify, and distribute `cooked-validators` under
+the terms of the MIT license. We provide `cooked-validators` as a
+research prototype under active development, and it comes _as is_ with
+no guarantees whatsoever. Check the [license](LICENSE) for details.
 
 ## How to integrate `cooked-validators` in a project
 
-This guide shows you how to use `cooked-validators` in a haskell project
-using [Cabal](https://cabal.readthedocs.io/en/stable/)
-to create and validate a simple transaction.
-
-Before using `cooked-validators`, you need
-- [GHC](https://www.haskell.org/ghc/download_ghc_8_10_7.html) version 8.10.7
-- [Cabal](https://www.haskell.org/cabal)
+To use `cooked-validators`, you need
+- [GHC](https://www.haskell.org/ghc/download_ghc_9_6_5.html) version 9.6.5
+- [Cabal](https://www.haskell.org/cabal) version 3.10 or later
 
-1. If you have no constraint on the version of `plutus-apps`, copy the file
-   [`cabal.project`](./cabal.project) to your project and
-   [adapt](https://cabal.readthedocs.io/en/stable/cabal-project.html#specifying-the-local-packages)
-   the `packages` stanza.
+1. `cooked-validators` depends on
+[cardano-haskell-packages](https://github.com/input-output-hk/cardano-haskell-packages)
+to get cardano-related packages and on
+[cardano-node-emulator](https://github.com/IntersectMBO/cardano-node-emulator)
+directly. If you have no constraint on the version of this package,
+copy the file [`cabal.project`](./cabal.project) to your project and
+[adapt](https://cabal.readthedocs.io/en/stable/cabal-project.html#specifying-the-local-packages)
+the `packages` stanza.
+   
 2. Add the following stanza to the file `cabal.project`
    ```cabal.project
    source-repository-package
      type: git
      location: https://github.com/tweag/cooked-validators
-     tag: v3.0.0
+     tag: myTag
      subdir:
-       cooked-validators
+       .
    ```
-3. Make your project
+   where `myTag` is either a commit hash in the repo, or a tag, such as v4.0.0
+   
+## An example of usage of `cooked-validators`
+   
+1. Make your project
    [depend](https://cabal.readthedocs.io/en/stable/getting-started.html#adding-dependencies)
    on `cooked-validators` and `plutus-script-utils`
+   
 3. Enter a Cabal read-eval-print-loop (with `cabal repl`)
```

Comment:
```suggestion
3. Enter a REPL (with `cabal repl`)
```

Reply:
For some reason, this sentence was there from the beginning so I'm for keeping it this way. That's how I learned what "repl" stands for, maybe others can enjoy this too.